CRM/Core/BAO/Mapping.php

```diff
@@ -990,6 +990,80 @@ public static function addComponentFields(&$fields, $mappingType, $exportMode) {
     return $compArray;
   }
 
+  /**
+   * Get the parameters for a mapping field in a saveable format from the quickform mapping format.
+   *
+   * @param array $params
+   * @param int $mappingId
+   * @param array $v
+   * @param string $k
+   * @param string $key
+   * @param int $colCnt
+   *
+   * @return array
+   */
+  protected static function getMappingParams($params, $mappingId, $v, $k, $key, int $colCnt): array {
```

Comment:
This is the first time I saw this syntax with a colon and return type. Neat. But it only works in php 7.

    CRM_Upgrade_Form::MINIMUM_PHP_VERSION = 5.6

Reply:
hmm we HAVE dropped php 5.6 support though - maybe we should update that